### PR TITLE
Sequence diagram performance

### PR DESCRIPTION
**Pull request by lredor**
Signed-off-by: Nathalie Lepine <nathalie.lepine@obeo.fr>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/InteractionRefreshExtension.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+   * Cache for ChildExecution -> ParentExecution and InteractionState -> ParentExecution.
+   */
+  private static Map<EObject, Collection<EObject>> stateElementToContainerCache = new ConcurrentHashMap<>();
+
+  /**
+   * Cache for StateFragment -> ParentExecution and StateFragment -> ParentInstanceRole.
+   */
+  private static Map<EObject, Collection<EObject>> executionElementToContainerCache = new ConcurrentHashMap<>();
+
+  /**
```
Comment (mPorhel) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/InteractionRefreshExtension.java:49]: I wonder if we could not have a single collection/cache here.

File: core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/InteractionRefreshExtension.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+   * Cache for CommunicationItem -> sending End.
+   */
+  private static Map<CommunicationItem, AbstractEnd> communicationItemToSendingEndCache = new ConcurrentHashMap<CommunicationItem, AbstractEnd>();
+
+  /**
+   * Cache for CommunicationItem -> receiving End.
+   */
+  private static Map<CommunicationItem, AbstractEnd> communicationItemToReceivingEndCache = new ConcurrentHashMap<CommunicationItem, AbstractEnd>();
+
```
Comment (mPorhel) [inline core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/refresh/extension/InteractionRefreshExtension.java:58]: Those maps seems unused

**Review by mPorhel — COMMENTED** [review]
Hi Nathalie, 
I get good result with your commit but we need to discuss about further changes.

**Comment by mPorhel** [comment]
This PR shoudl be rejected / abandoned, it has been replaced by https://github.com/eclipse/capella/pull/381

**Comment by pdulth** [comment]
Merged through #381

**Pull request closed without merging**